index.js

```diff
@@ -24,6 +25,17 @@ module.exports = function(dep, filename, directory) {
 
   if (!depExt) {
     filepath += fileExt;
+  } else {
+    // If a dependency starts with a period AND it doesn't already end
+    // in .js AND doesn't use a custom plugin, add .js back to path.
+    if (fileExt === '.js' && !dep.endsWith('.js') && dep.indexOf('!') < 0) {
```

Comment:
It's great that you supplied systemjs support. This logic won't hold for webpack though, where `require('./styles/foo.css')` is a valid call within a JS file.